inspirehep/modules/authors/receivers.py

```diff
@@ -32,6 +36,60 @@
 from .utils import author_tokenize
 
 
+def _query_beard_api(full_names):
+    """Query Beard API.
+
+    This method allows for computation of phonetic blocks
+    from a given list of strings.
+    """
+    headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json"
+    }
+
+    text_endpoint = "{base_url}/text/phonetic_blocks".format(
+        base_url=current_app.config.get('BEARD_API_URL')
+    )
+
+    response = requests.post(
+        url=text_endpoint,
+        headers=headers,
+        data=json.dumps({'full_names': full_names})
+    )
+
+    return response.json()['phonetic_blocks']
+
+
+@before_record_insert.connect
+@before_record_update.connect
+def assign_phonetic_block(sender, *args, **kwargs):
+    """Assign phonetic block to each signature.
+
+    The method extends the given signature with a phonetic
+    notation of the author's full name, based on
+    nysiis algorithm. The phonetic block is assigned before
+    the signature is indexed by an Elasticsearch instance.
+    """
+    if current_app.config.get('BEARD_API_URL'):
+        authors = sender.get('authors', [])
+        authors_map = {}
+        full_names = []
```

Comment:
This will just contain the `.keys()` of `authors_map`, so it's not really needed.